main.py

```diff
@@ -4,64 +4,130 @@
 from falkordb import FalkorDB
 import pandas as pd
 
-def export_graph(graph_name, host, port):
+def export_graph(graph_name, host, port, username=None, password=None, split_by_type=True):
     # Connect to FalkorDB
-    db = FalkorDB(host=host, port=port)
+    connection_kwargs = {"host": host, "port": port}
+    if username:
+        connection_kwargs["username"] = username
+    if password:
+        connection_kwargs["password"] = password
+    
+    db = FalkorDB(**connection_kwargs)
     g = db.select_graph(graph_name)
 
     # Export Nodes by Label
-    nodes_result = g.ro_query("MATCH (n) RETURN ID(n), labels(n), properties(n)")
+    print("🔄 Fetching nodes...")
+    skip = 0
+    batch_size = 10000
     nodes_by_label = defaultdict(list)
-
-    for record in nodes_result.result_set:
-        node_id = record[0]
-        labels = record[1]
-        props = record[2] or {}
-
-        # Handle nodes with multiple labels or no labels
-        if labels:
-            for label in labels:
+    total_nodes = 0
+
+    while True:
+        nodes_result = g.ro_query(
+            f"MATCH (n) RETURN ID(n), labels(n), properties(n) SKIP {skip} LIMIT {batch_size}"
```

Comment:
It will be faster if instead of skip filter wit id(n) > last id

Reply:
id is not necessarily sortable... I think current approach is "good enough"...